src/main/java/org/humancellatlas/ingest/notifications/NotificationService.java

```diff
@@ -0,0 +1,78 @@
+package org.humancellatlas.ingest.notifications;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.stream.Stream;
+import lombok.AllArgsConstructor;
+import org.humancellatlas.ingest.notifications.exception.DuplicateNotification;
+import org.humancellatlas.ingest.notifications.model.Checksum;
+import org.humancellatlas.ingest.notifications.model.Notification;
+import org.humancellatlas.ingest.notifications.model.NotificationRequest;
+import org.humancellatlas.ingest.notifications.model.NotificationState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Component;
+
+
+@Component
+@AllArgsConstructor
+public class NotificationService {
+
+  private final NotificationRepository notificationRepository;
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  public Notification createNotification(NotificationRequest notificationRequest) {
+    try {
+      Notification notification = Notification.buildNew()
+                                              .content(notificationRequest.getContent())
+                                              .metadata(notificationRequest.getMetadata())
+                                              .checksum(notificationRequest.getChecksum())
+                                              .build();
+
+      return this.notificationRepository.save(notification);
+    } catch (DuplicateKeyException e) {
+      String checksumValue = notificationRequest.getChecksum().getValue();
+      String id = this.notificationRepository.findByChecksum_Value(checksumValue)
```

Comment:
The ID is being retrieved here, given the checksum

Reply:
I see, ok. But is it necessary to retrieve the id? I think the checksum would be enough information to find the conflict (if needed), especially since it's guaranteed to be unique.